draft-holmberg-mmusic-ice-premature.xml

```diff
@@ -58,35 +58,33 @@
         for Network Address Translator (NAT) traversal for UDP-based communication.
       </t>
       <t>
-        When an endpoint receives the initial set of candidates from the remote peer, it starts to
-        process them. As part of the processing, one or more candidates might get discarded. There
-        are a number of reasons why a candidate is discared, e.g., because it represents an IP version
-        that is not supported by the endpoint, or if the connection-address associated with the candidate
-        is represented by a Fully Qualified Domain Name (FQDN) that the endpoint is not able to resolve.
-        Similairly, as a candidate by definition represents a single connection-address, if a FQDN
-        resoles into multiple connection-addresses the associated candidate needs to be discared.
+        Typically, when using ICE, endpoints will exchange candidate addresses,
+        form a list of candidate pairs, and then test each candidate pair to see
+        if connectivity can be established. If the test for a given pair fails,
+        it is marked accordingly, and if all pairs have failed, the overall
+        ICE process is considered to have failed.
       </t>
```

Comment:
I think it would be useful to indicate that a test failure can occur before the connectivity test message has even been sent, e.g., due to FQDN resolve failure.

Reply:
I tend to think this is overly specific for the introduction, since the document explains this particular case later.

Reply:
Sure, but since we know that FQDNs are likely to become more common, due to mDNS, I see no harm in mentioning it. Roman also mentioned unsupported IP-version as one reason for discarding candidates, and I don't think it is currently covered in 8445.

Reply:
OK, will add a short note on this.

Reply:
I added some text to point out that there could be no candidate pairs in certain cases.